### PR TITLE
Dk modkit

### PR DESCRIPTION
**Pull request by dkessner**
This change adds a process to run the modkit command line tool for generating bedMethyl files from modBAM files.
https://github.com/nanoporetech/modkit

There is a new file modules/local/modkit.nf, and a single change to workflows/sarek.nf to call the MODKIT process after merging bam files.

I tested this change on a small merged bam file during development, and I ran the pipeline successfully on STATE 035 data.

Here is the repo for the Docker file I created for modkit:
https://github.com/eitm-org/modkit_docker

And it's published here:
https://github.com/orgs/eitm-org/packages/container/package/modkit

### CODE REVIEW

**Review by xingyaoc — CHANGES_REQUESTED** [review]
The module looks good. A couple nits.
1. For exposing args, can you please add a config file for this process ([example](https://github.com/eitm-org/sarek/blob/master/conf/modules/modules.config))
2. Is there an option that can be passed in from the `nextflow run` command that allows you to skip modkit and/or only run modkit without running downstream workflows (ie variant calling and markduplicates)? If not can you please add that logic into sarek.nf and the config ([example](https://github.com/eitm-org/sarek/blob/master/conf/modules/clairs.config#L49C66-L49C66))?

**Comment by xingyaoc** [comment]
> The module looks good. A couple nits.
> 
> 1. For exposing args, can you please add a config file for this process ([example](https://github.com/eitm-org/sarek/blob/master/conf/modules/modules.config))
> 2. Is there an option that can be passed in from the `nextflow run` command that allows you to skip modkit and/or only run modkit without running downstream workflows (ie variant calling and markduplicates)? If not can you please add that logic into sarek.nf and the config ([example](https://github.com/eitm-org/sarek/blob/master/conf/modules/clairs.config#L49C66-L49C66))?

For 2, one option I see is to use `--skip-tools`, this might require the least code change.  https://github.com/eitm-org/sarek/blob/master/nextflow_schema.json#L94-L99

**Comment by dkessner** [comment]
Not nits!  Config file is the right way to do it (though Nextflow config still seems convoluted to me).  I added conf/modules/modkit.config, which is now included from workflows/sarek.nf.

I needed to make one change to nextflow_schema.json in order for "--tools modkit" to be parsed correctly.  Note: I edited this manually to get it working.  I believe the schema is supposed to be generated with the nf-core cli tool "nf-core schema ...".  Let me know if you want me to spend more time looking into this (schema updates + nf-core cli tool).

Command line must now include:
--tools "clairs,modkit"


Reference for nf-core schema:
https://nf-co.re/tools#pipeline-schema

File: modules/local/modkit.nf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    prefix = task.ext.prefix ?: "${meta.id}"
+    bed = "${prefix}.bed"
+    summary = "${prefix}.summary"
+    modkit_log = "${prefix}.log"
+
+    // modkit command line options:
+    //  https://nanoporetech.github.io/modkit/
+
+    modkit_args = " --ref $fasta"
```
Comment (xingyaoc) [inline modules/local/modkit.nf]: ext.args from modkit.config should be passed into here right? Example https://github.com/eitm-org/sarek/blob/master/modules/local/clairs.nf#L32 and https://github.com/eitm-org/sarek/blob/master/conf/modules/clairs.config#L50

**Review by xingyaoc — CHANGES_REQUESTED** [review]

File: modules/local/modkit.nf (hunk 1 above)
Comment (dkessner) [reply modules/local/modkit.nf]: Yep, my bad -- fixed now.

**Comment by xingyaoc** [comment]
The changes to config look good to me, please make sure to test the latest change with `--tools modkit` `--tools clairs` and `--tools modkit,clairs`. Also, for modkit, they seem to be releasing new [updates](https://github.com/nanoporetech/modkit/releases/tag/v0.1.12) regularly. Is is possible to specify the version number in your [docker image](https://github.com/eitm-org/modkit_docker/blob/main/Dockerfile) and similarly in the modkit nextflow module?

**Comment by dkessner** [comment]
Good idea to tag/pin the versions.  

I made the changes in eitm-org/modkit_docker to pin the version of modkit at 0.1.12:
https://github.com/eitm-org/modkit_docker

I also added a tag for the Docker image:
https://github.com/orgs/eitm-org/packages/container/modkit/116516916?tag=1.0

And made small change in modkit.nf to use the tagged version ghcr.io/eitm-org/modkit:1.0.

My tests with --tools clairs and --tools "clairs,modkit" work, but --tools modkit fails with error "Missing workflow output parameter: clairs_vcf_normal_germline".  I hadn't tested the pipeline without "--tools clairs" before -- this may be a separate issue.

I'm currently investigating -- stay tuned.

**Comment by dkessner** [comment]
This one was tricky to debug -- Nextflow offered very little guidance.

The variant calling code was controlled by the condition:
  if (params.tools) { ...
https://github.com/eitm-org/sarek/blob/master/workflows/sarek.nf#L996

Running the pipeline with '--tools modkit' makes this condition true, but we don't want the variant caller to run in this case.  

I've changed the condition to check specifically for "clairs" in the --tools string.

I've tested that the pipeline runs successfully with:
1) no --tools arg
2) --tools clairs
3) --tools modkit
4) --tools "clairs,modkit"

**Review by xingyaoc — APPROVED** [review]
:shipit:

**Pull request merged**